pages/Out-of-band resource load (HTTP).md

```diff
@@ -0,0 +1,38 @@
+---
+
+layout: col-sidebar
+title: Out-of-band resource load (HTTP)
+author: Rishu Ranjan
+contributors: 
+permalink: /Out-of-band resource load (HTTP)
+tags: attack, Out-of-band resource load (HTTP), OBRL
+
+---
+
+
+# Description
+
+Out-of-band resource load arises when it is possible to induce an application to fetch content from an arbitrary external location, and incorporate that content into the application's own response(s). The ability to trigger arbitrary out-of-band resource load does not constitute a vulnerability in its own right, and in some cases might even be the intended behavior of the application. However, in many cases, it can indicate a vulnerability with serious consequences.
+
+The ability to request and retrieve web content from other systems can allow the application server to be used as a two-way attack proxy. By submitting suitable payloads, an attacker can cause the application server to attack, or retrieve content from, other systems that it can interact with. This may include public third-party systems, internal systems within the same organization, or services available on the local loopback adapter of the application server itself. Depending on the network architecture, this may expose highly vulnerable internal services that are not otherwise accessible to external attackers.
+
+Additionally, the application's processing of web content that is retrieved from arbitrary URLs exposes some important and non-conventional attack surface. An attacker can deploy a web server that returns malicious content, and then induce the application to retrieve and process that content. This processing might give rise to the types of input-based vulnerabilities that are normally found when unexpected input is submitted directly in requests to the application. The out-of-band attack surface that the application exposes should be thoroughly tested for these types of vulnerabilities.
+
+> Attack Type: Server-Side
+
+# Audit Guideline
+
+1) Capture the base request of the vulnerable website in the burp community and send the request to the repeater.
+2) Change the request path to any internal or external URL. For eg. change the request path to https://www.attacker.com.
+3) The application will include the response of https://www.attacker.com with the response headers of vulnerable website.
```

Comment:
```suggestion
3) The application will include the response of `https://www.attacker.com` with the response headers of vulnerable website.
```

> with the response headers of vulnerable website.

It will?
Why is this relevant, it hasn't been mentioned till this point.